README.md

```diff
@@ -54,6 +63,403 @@ npm install @dojo/shim
+
+#### Registering Additional Routes
+
+Additional routing configuration can be registered with a router instance, either from the root or by specifying an existing outlet name.
+
+```ts
+const additionalRouteConfig = [
+	{
+		path: 'extra',
+		outlet: 'extra'
+	}
+];
+
+// Will register the extra routes from the route
+router.register(additionalRouteConfig);
+
+// Will register the extra routes from `foo` outlet
+router.register(additionalRouteConfig, 'foo');
+```
+
+If the outlet is not found then an error is thrown.
+
+#### Router Context Injection
+
+To make the `router` instance available to the created outlets and other routing components (such as [`Link`](#link)) Routing leverages a core Dojo 2 widget-core concept [Injecting State](https://github.com/dojo/widget-core/blob/master/README.md#injecting-state). The custom injector `RouterInjector` needs to be defined in a registry available to the components for an known `key`.
+
+All routing components by default use the exported `RouterInjector#routerKey` as the key for the injected `router` context.
+
+```ts
+import { registry } from '@dojo/widget-core/d'
+import { RouterInjector, routerKey } from '@dojo/routing/RouterInjector';
+
+registry.define(routerKey, Injector(RouterInjector, router));
+```
+
+`RouterInjector` module exports a useful helper function, `registerRouterInjector` that combines the instantiation of a `Router` instance, registering route configuration and defining the `RouterInjector` and returns the `router` instance.
+
+```ts
+import { registerRouterInjector } from '@dojo/routing/RoutingInjector';
+
+const router = registerRouterInjector(config);
+```
+
+The defaults can be overridden using the `registry`, `history` and `key` arguments
+
+```ts
+import { WidgetRegistry } from '@dojo/widget-core';
+import { registerRouterInjector } from '@dojo/routing/RoutingInjector';
+import MemoryHistory from './history/MemoryHistory';
+
+const customRegistry = new WidgetRegistry();
+const history = new MemoryHistory();
+
+const router = registerRouterInjector(config, customRegistry, history, 'custom-router-key');
+```
+
+The final thing to do is call `router.start()` to the start the `router` instance.
+
+#### Link Component
+
+The `Link` component it a wrapper around an `a` DOM element that enables consumers to specify an `outlet` to create a link to. It is also possible to use a static route by setting the `isOutlet` property to `false`.
+
+If the generated link requires specific path or query params that are not in the route then they can be passed via the `params` property.
+
+```ts
+import { Link } from '@dojo/routing/Link';
+
+render() {
+	return v('div', [
+		w(Link, { to: 'foo', params: { foo: 'bar' }}, [ 'Link Text' ]),
+		w(Link, { to: '#/static-route', isOutlet: false, [ 'Other Link Text' ])
+	]);
+}
+```
+
+All the standard `VirtualDomProperties` are available for the `Link` component as they would be creating an `a` DOM Element using `v()`.
```

Comment:
Should probably make it clear that the standard `VirtualDomProperties` and `v()` are from @dojo/widget-core?